filament/backend/src/opengl/OpenGLProgram.cpp

```diff
@@ -211,6 +211,9 @@ void OpenGLProgram::updateSamplers(OpenGLDriver* const gld) const noexcept {
         size_t const binding = usedBindingPoints[i];
         assert_invariant(binding < Program::SAMPLER_BINDING_COUNT);
         auto const * const sb = samplerBindings[binding];
+        if (!sb) { // A sampler may not need to be used, although it's declared in the shader
+            continue;
+        }
```

Comment:
We should never allow this because it's not allowed in Vulkan and Metal. Also there is an assert for this just below.
If there are scenarios where this happens, we should fix those.

Reply:
Ok I see, but it may happen in OpenGL.

Reply:
If it happens we need to understand why and not allow it to happen.